apps/rush-lib/src/logic/taskRunner/TaskRunner.ts

```diff
@@ -428,13 +428,16 @@ export class TaskRunner {
         }
 
         if (task.writer) {
-          let stderr: string = task.writer.getStdError();
-          if (stderr && (task.status === TaskStatus.Failure || task.status === TaskStatus.SuccessWithWarning)) {
-            stderr = stderr.split(os.EOL)
+          const stderr: string = task.writer.getStdError();
+          const shouldPrintDetails: boolean =
+            task.status === TaskStatus.Failure || task.status === TaskStatus.SuccessWithWarning;
+          let details: string = stderr ? stderr : task.writer.getStdOutput();
```

Comment:
Will `getStdOutput()` potentially return a much larger amount of text than `stderr`?  If so, should we limit it to a maximum number of lines?

Reply:
Reasoning was: "since we print all of stderr without limit, might as well do the same with stdout". But this morning I agree it's not great, as stdout is very often considerably larger in practice, so practically speaking it makes sense to make that effort for stdout. Thanks, will see how I can improve.

Reply:
I'm thinking we might want the first and last N lines here. Do you prefer bespoke code within `TaskRunner.ts`, or should I add a reusable method in `Text.ts`?

Reply:
Generally we tend to keep stuff scoped as privately as possible at first.  We can promote it to being shared (or eventually public API) later when we encounter some other component that actually needs it. The rationale is that if we don't have at least one concrete example of another caller, then we're probably not in a great situation to be designing the API yet.